zerver/views/message_edit.py

```diff
@@ -143,13 +143,14 @@ def validate_can_delete_message(user_profile: UserProfile, message: Message) ->
     return
 
 
+@transaction.atomic
 @has_request_variables
 def delete_message_backend(
     request: HttpRequest,
     user_profile: UserProfile,
     message_id: int = REQ(converter=to_non_negative_int, path_only=True),
 ) -> HttpResponse:
-    message, ignored_user_message = access_message(user_profile, message_id)
+    message, ignored_user_message = access_message(user_profile, message_id, lock_message=True)
     validate_can_delete_message(user_profile, message)
     try:
         do_delete_messages(user_profile.realm, [message])
```

Comment:
What will happen if someone had locked a UserMessage row when we call this?  I guess what I'd hope for is that we just finish the other transaction happily and then run this.  (In which case it should be illegal to lock a UserMessage if you're later going to want to lock the Message within that transaction).

One possible theory for not needing to think about this sort of question is that we should just always lock the message object (and never lock UserMessage objects).